src/plugins/intel_cpu/src/nodes/random_uniform.hpp

```diff
@@ -0,0 +1,120 @@
+        int64_t  i64;
+    };
+
+    RandomUniform(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context);
+
+    void getSupportedDescriptors() override;
+
+    void initSupportedPrimitiveDescriptors() override;
+
+    bool needPrepareParams() const override;
+
+    void prepareParams() override;
+
+    void execute(dnnl::stream strm) override;
+
+    void executeDynamicImpl(dnnl::stream strm) override;
+
+    bool isExecutable() const override;
+
+    void createPrimitive() override;
+
+    bool created() const override;
+
+    bool canBeInPlace() const override { return false; }
+
+    static bool isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept;
+
+    std::string getPrimitiveDescriptorType() const override;
+
+protected:
+    bool needShapeInfer() const override;
+
+private:
+    void computeStl(void* out, size_t work_amount);
+
+    std::pair<uint64_t, uint64_t> computePhilox(void* out, size_t work_amount, const std::pair<uint64_t, uint64_t>& prev_state);
+
+    template <typename T, typename DISTR_TYPE>
+    void generateData(DISTR_TYPE distribution, void* out, size_t work_amount);
+
+    void initOutShape(VectorDims& dst, const void* src, const element::Type& shape_type, size_t len);
+
+    void initEdgeValues(OutputType& dst, const void* src, const element::Type& output_type);
+
+    enum { SHAPE = 0, MIN_VAL, MAX_VAL };
+    enum AlgoType { STL, PHILOX };
+
+    bool m_const_inputs[3] = {false, false, false};
+
+    ov::element::Type m_output_prc;
+    uint64_t m_global_seed = 0lu;
+    uint64_t m_op_seed = 0lu;
+    std::pair<uint64_t, uint64_t> m_state {0lu, 0lu};
+
+    VectorDims m_out_shape = {};
+    uint64_t m_out_el_num = 1lu;
+    OutputType m_min_val;
+    OutputType m_max_val;
+    AlgoType m_algo = PHILOX;
+
+    std::default_random_engine m_generator;
+
+    struct ThreadParams {
+        uint64_t work_amount = 0lu;
+        uint64_t dst_shift = 0lu;
+        uint64_t n_shift = 0lu;
+
+        uint64_t step = 0lu;
+        uint64_t step_b = 0lu;
+        int64_t work_amount_b = 0;
+    };
+
+    uint64_t m_threads_num = 0lu;
+    std::vector<ThreadParams> m_thread_params;
+
+    ///// PHILOX constants /////
+
+    // Determines how many sequence elements of RNG sequence are skipped between runs.
+    // Can be any positive value, 256 is chosen for parity with Tensorflow.
+    static constexpr uint64_t SKIP_CONST = 256lu;
+
+    // Philox algorithm returns 4 elements of RNG sequence per each invocation
+    static constexpr uint64_t PHILOX_GROUP_SIZE = 4lu;
+
+    // Output elements number threshold to execute on one thread.
+    static constexpr uint64_t PHILOX_PARALLEL_EXECUTION_THRESHOLD = 1000lu;
+
+    uint64_t m_skip_count = 0lu;
+    /////////////////////////////////////////////////////////////////////////////////
+
+    std::shared_ptr<kernel::JitKernel<kernel::RandomUniformCompileParams, kernel::RandomUniformCallArgs>> m_jit_kernel;
```

Comment:
What's about to add `using RandomUniformKernel = kernel::JitKernel<kernel::RandomUniformCompileParams, kernel::RandomUniformCallArgs>;`?

Reply:
Moreover, I'm not sure that it will be built on ARM machine since `jit_kernel` is not compiled on these machines

Reply:
`using RandomUniformKernel` was added.
About the ARM: there is another execution branch without the kernel.

Reply:
I understand that we have another execution branch. But `JitKernel` is defined in [jit_kernel_base.hpp](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/kernels/x64/jit_kernel_base.hpp). But this file is not [compiled](https://github.com/openvinotoolkit/openvino/blob/a13cc8c733c443bf51091e3cf8c9bddbeb11ccef/src/plugins/intel_cpu/CMakeLists.txt#L82) on ARM. I think that this code won't be compiled on ARM. Maybe I missed something?

Reply:
As I can see, it's built successfully. But I think we should avoid these situations with x64 code compilation, shouldn't we?

@dmitry-gorokhov correct me please if I'm wrong.